spec/controllers/doi_requests_controller_spec.rb

```diff
@@ -1,19 +1,166 @@
 require 'rails_helper'
 
+#doi_requests_controller_spec
 RSpec.describe DoiRequestsController, type: :controller do
+  # render_views
+  # https://github.com/rspec/rspec-rails recommends not using render_views
+  
 
-  describe "GET #index" do
-    it "returns http success" do
-      get :index
-      expect(response).to have_http_status(:success)
+  # before do
+  #   allow_any_instance_of(User).to receive(:groups).and_return([])
+  # end
+
+  let(:user) {FactoryGirl.create(:user) }
+  let(:admin) {FactoryGirl.create(:user, :with_admin_role) }
+
+
+  describe '#index' do
+    let(:doi_request) {FactoryGirl.create(:doi_request)}
+
+    context "for normal users" do
+      before {sign_in user}
+
+      it 'should redirect' do
+        get :index
+        expect(response).not_to have_http_status(200)
+      end
     end
+
+    context "for admin users" do
+      before {sign_in admin}
+      it 'should render index for admin users' do
+        get :index
+        expect(assigns[:doi_requests]).to eq [doi_request]
+      end
+    end
+
   end
 
-  describe "GET #show" do
-    it "returns http success" do
-      get :show
-      expect(response).to have_http_status(:success)
+
+  describe '#pending' do
+    let(:doi_request) {FactoryGirl.create(:doi_request)}
+
+    it 'should render pending doi requests for admin users' do
+      sign_in admin
+      get :pending
+      expect(assigns[:doi_requests]).to eq [doi_request]
     end
   end
 
+
+  describe '#create' do
+    before do
+      sign_in user
+      @collection = Collection.create(title: "My collection",
+                        description: "My incredibly detailed description of the collection"
+                        ) { |c| c.apply_depositor_metadata(user) }
+    end
+
+    it "creates a Doi Request" do
+      expect do
+        post :create, asset_id: @collection.id, asset_type: "Collection"
+      end.to change { DoiRequest.count }.by(1)
+    end
+  end
+
+
+  describe "#mint" do
+    let(:doi_request) do 
+      c = FactoryGirl.build(:collection)
+      c.apply_depositor_metadata(user.user_key)
+      c.save
+      c[:identifier] << 'doi:pending'
+      doi_request = DoiRequest.new(asset_id: c.id)
+      doi_request.save
+      c.update_attributes({:identifier => c[:identifier]})
+      doi_request
+    end
+
+    before do
+      sign_in admin
+    end
+
+    it "mints a doi_request" do
+      patch :mint_doi, id: doi_request
+      expect(assigns[:doi_request].ezid_doi).to match /doi\:.*\/.*/
+      expect(assigns[:doi_request].ezid_doi).not_to match /pending/
+    end
+  end
+
+  describe '#show' do
+    # let(:create_collection) do 
```

Comment:
This was a failed attempt at implementation. It was later implemented with much simpler code!